ratis-netty/src/main/java/org/apache/ratis/netty/server/DataStreamManagement.java

```diff
@@ -383,8 +409,20 @@ static void sendDataStreamException(Throwable throwable, DataStreamRequestByteBu
     }
   }
 
-  void cleanUpOnChannelInactive(ClientInvocationId key) {
-    Optional.ofNullable(streams.remove(key)).ifPresent(removed -> removed.getLocal().cleanUp());
+  void cleanUpOnChannelInactive(ChannelHandlerContext ctx, TimeDuration channelInactiveGracePeriod) {
+    // Delayed memory garbage cleanup
+    Optional.ofNullable(requestTracker.removeAll(ctx.channel().id())).ifPresent(ids -> {
+      final String streamIds = Arrays.toString(ids.toArray());
+      LOG.info("server {} is disconnect, cleanup clientInvocationIds={}", ctx.channel(), streamIds);
+      TimeoutExecutor.getInstance().onTimeout(channelInactiveGracePeriod,
+          () -> {
+            for (ClientInvocationId clientInvocationId : ids) {
+              Optional.ofNullable(streams.remove(clientInvocationId)).ifPresent(removed ->
+                  removed.getLocal().cleanUp());
+            }
+          },
```

Comment:
It will be easier to read if it is refactored out
```java
  void cleanUp(Set<ClientInvocationId> ids) {
    for (ClientInvocationId clientInvocationId : ids) {
      Optional.ofNullable(streams.remove(clientInvocationId))
          .map(StreamInfo::getLocal)
          .ifPresent(LocalStream::cleanUp);
    }
  }

  void cleanUpOnChannelInactive(ChannelId channelId, TimeDuration channelInactiveGracePeriod) {
    // Delayed memory garbage cleanup
    Optional.ofNullable(channels.remove(channelId)).ifPresent(ids -> {
      LOG.info("Channel {} is inactive, cleanup clientInvocationIds={}", channelId, ids);
      TimeoutExecutor.getInstance().onTimeout(channelInactiveGracePeriod, () -> cleanUp(ids),
          LOG, () -> "Timeout check failed, clientInvocationIds=" + ids);
    });
  }
```